lisp/ess-r-mode.el

```diff
@@ -2219,6 +2154,42 @@ otherwise nil."
       (undo-boundary))))
 
 
+
+;; Create functions that can be called for running different versions
+;; of R.
+;; FIXME: Should be set in ess-custom
+(setq ess-rterm-version-paths
+      (ess-flatten-list
+       (ess-uniq-list
+        (if (not ess-directory-containing-R)
+            (if (getenv "ProgramW6432")
+                (let ((P-1 (getenv "ProgramFiles(x86)"))
+                      (P-2 (getenv "ProgramW6432")))
+                  (nconc
+                   ;; Always 32 on 64 bit OS, nil on 32 bit OS
+                   (ess-find-rterm (concat P-1 "/R/") "bin/Rterm.exe")
+                   (ess-find-rterm (concat P-1 "/R/") "bin/i386/Rterm.exe")
+
+                   ;; Keep this both for symmetry and because it can happen:
+                   (ess-find-rterm (concat P-1 "/R/") "bin/x64/Rterm.exe")
+
+                   ;; Always 64 on 64 bit OS, nil on 32 bit OS
+                   (ess-find-rterm (concat P-2 "/R/") "bin/Rterm.exe")
+                   (ess-find-rterm (concat P-2 "/R/") "bin/i386/Rterm.exe")
+                   (ess-find-rterm (concat P-2 "/R/") "bin/x64/Rterm.exe")))
+              (let ((PF (getenv "ProgramFiles")))
+                (nconc
+                 ;; Always 32 on 32 bit OS, depends on 32 or 64 process on 64 bit OS
+                 (ess-find-rterm (concat PF "/R/") "bin/Rterm.exe")
```

Comment:
I think ess-find-rterm should be refactored into handling these 3 cases internally. I think the new logic here doesn't match with old logic in ess-find-rterm, this is why Henric having problems on the mailing list.

Reply:
I find the situation on Windows pretty confusing TBH. It probably doesn't help that I don't have a windows machine to test on.

Have you taken a look at this: https://github.com/jabranham/ESS/commit/43501f19167d94dbe8ef5a604bcafe53183ec8e3 ? I think that would solve Henric's problem but I haven't heard back from him yet.

Reply:
Yeh, it's a bit puzzling but you are probably right with that lexical binding. That builder was the only change in logic as far as I can judge.

 I think all of us have to setup a windows in a virtual box and test it there at least occasionally. 

> Also in my mind that's a separate PR since this one deals with separating R/S runner definition 

Indeed.